### PR TITLE
Fix Query Errors in TableAnalysis::tableExists()

### PR DESCRIPTION
**Pull request by tn3rb**
While activating a bunch of add-ons on https://barista.eventespresso.com/ I noticed there were a bunch of db errors that occurred:

ScreenShot_20201013130401

which all originated from the `TableAnalysis::tableExists()` method.

That method was essentially running a query to return a value from the desired table and then checking for wpdb errors to determine if the table exists or not. ie: no table === missing table error

This PR replaces the logic in `tableExists()` with essentially the same code that WP core uses in their [maybe_create_table() function](https://developer.wordpress.org/reference/functions/maybe_create_table/#source)  which does not result in any errors because it is not intentionally relying on errors to determine if a table exists or not.

### CODE REVIEW

**Review by manzoorwanijk — APPROVED** [review]
Looks good

File: core/services/database/TableAnalysis.php
Comment (knazart) [inline core/services/database/TableAnalysis.php:75]: Most likely phpcs won't like this.

```suggestion
        $query = $wpdb->prepare('SHOW TABLES LIKE %s', $wpdb->esc_like($table_name));
        return $wpdb->get_var($query) === $table_name;
```

**Review by knazart — COMMENTED** [review]
Looks good, but might be getting a notice from phpcs.

File: core/services/database/TableAnalysis.php
Comment (tn3rb) [reply core/services/database/TableAnalysis.php:75]: yup saw that thnx...
this change also opened a can of worms with the unit tests because they were all set up to expect errors to be thrown during activation, so I'm currently trying to fix those issues right now

**Comment by tn3rb** [comment]
Had to revert back to what Michael had done previously in the `tableExists()` method because I could not get things to work properly. I also found some old issues on codebase where Michael had switched away from using "SHOW TABLES LIKE" queries because they were having a performance impact on EventSmart.
see: https://events.codebasehq.com/projects/event-espresso/tickets/9710

There was a bunch of formatting and code cleanup that I had done while working on this that I decided to push anyways so that this didn't turn out to be a total loss.

File: core/services/database/TableAnalysis.php
Comment (manzoorwanijk) [inline core/services/database/TableAnalysis.php:109]: 😄

File: core/helpers/EEH_Activation.helper.php
Comment (knazart) [inline core/helpers/EEH_Activation.helper.php:1644]: So we still do these indents ? Asking because I remember when we first changed our coding standards we made a decision to drop this. So I usually don't do these indents.

File: tests/includes/EE_UnitTestCase.class.php
Comment (knazart) [inline tests/includes/EE_UnitTestCase.class.php:991]:
```suggestion
            $this->table_analysis = EE_Registry::instance()->create('TableAnalysis', [], true);
```

File: tests/testcases/core/libraries/plugin_api/EE_Register_Model_Extensions_Test.php
Comment (knazart) [inline tests/testcases/core/libraries/plugin_api/EE_Register_Model_Extensions_Test.php:179]:
```suggestion
    $this->initTableAnalysis();
		if ($table_name == 'esp_mock_attendee_meta' && ! $this->table_analysis->tableExists($table_name)) {
```

File: tests/testcases/core/libraries/plugin_api/EE_Register_Addon_Test.php
Comment (knazart) [inline tests/testcases/core/libraries/plugin_api/EE_Register_Addon_Test.php:80]:
```suggestion
        if (in_array($table_name, ['esp_new_addon_thing', 'esp_new_addon_attendee_meta'], true)
```

**Review by knazart — APPROVED** [review]
Looks good. A few things caught my eye (if you are neat picky 😃 ), but nothing to not approve this.

File: core/helpers/EEH_Activation.helper.php
Comment (tn3rb) [reply core/helpers/EEH_Activation.helper.php:1644]: oh my IDE is doing it automatically...
I can switch back... just have to find that setting...
unless you prefer aligned elements like this and want to switch over...
I can't remember who was opposed to this and/or what the reason might be to not do it!

File: core/services/database/TableManager.php
Comment (tn3rb) [reply core/services/database/TableManager.php:228]: i did that so that the resulting SQL was formatted in a SQL-ish way, but can revert to having this code all in one line (although the resulting SQL will still be multiline)

File: tests/includes/EE_UnitTestCase.class.php
Comment (tn3rb) [reply tests/includes/EE_UnitTestCase.class.php:991]: thnx will change

File: tests/testcases/core/libraries/plugin_api/EE_Register_Model_Extensions_Test.php
Comment (tn3rb) [reply tests/testcases/core/libraries/plugin_api/EE_Register_Model_Extensions_Test.php:179]: weird... my IDE should be formatting on paste AND on save, which obviously isn't happening...
also weird that his passed linting

good catch, will fix

File: tests/testcases/core/libraries/plugin_api/EE_Register_Addon_Test.php
Comment (tn3rb) [reply tests/testcases/core/libraries/plugin_api/EE_Register_Addon_Test.php:80]: thnx will fix

File: core/helpers/EEH_Activation.helper.php
Comment (knazart) [reply core/helpers/EEH_Activation.helper.php:1644]: PhpStorm ?
I'm not opposed of this at all. It's just that I'm not sure now if I should just do the same. Because a year ago I remember I was removing all those spaces.

File: core/services/database/TableManager.php
Comment (knazart) [reply core/services/database/TableManager.php:228]: Oh, right, ok.

Again, maybe not worth spending much time on this peace, but normally I would deal with this like so:

```suggestion
        $SQL = "CREATE TABLE {$table_name} ("
                . "{$create_sql}"
                . ") ENGINE={$engine} "
                . "{$collation}";
```

File: core/helpers/EEH_Activation.helper.php
Comment (tn3rb) [reply core/helpers/EEH_Activation.helper.php:1644]: Yes I'm using PhpStorm. What are you using?

My settings got reset recently because I just built a new system and reinstalled all of my software, so I'll have to find the setting for turning off that alignment. I do prefer having all of the array elements aligned because I think it looks neater and is easier to read, but I don't feel that strongly about it. If your IDE can do that automatically and you also prefer aligned elements, then let's change, but if you have to add those spaces manually, then don't bother. I will find the option in my settings and turn that off, because having to manually align things is not worth the benefit.

So I'll let you decide. Don't worry about what the option was in the past, just decide what you would prefer going forward.

Comment (knazart) [reply core/helpers/EEH_Activation.helper.php:1644]: > Yes I'm using PhpStorm. What are you using?

Same.

> I do prefer having all of the array elements aligned because I think it looks neater and is easier to read

Agree. It's just that one time when there's waay too many spaces on some short lines, then it looks a bit overwhelming. But yes, we can stick with this, I'm partly for it.
I'm using the EE settings from our GH repo, with a few of my own changes. But it never added those indents automatically, so I'll have to find that option and enable it myself.

**Review by knazart — APPROVED** [review]
🥇

**Comment by tn3rb** [comment]
killing this PR because the base branches have changed too much and had their own formatting applied.
not worth trying to sort things out now

**Pull request closed without merging**
